### PR TITLE
[FIX] link extension + popover closes when not in editable

### PR DESCRIPTION
**Pull request by Jinjiu96**

### CODE REVIEW

File: addons/html_editor/static/src/main/link/link_plugin.js [hunk 1]
```diff
@@ -204,6 +204,15 @@ export class LinkPlugin extends Plugin {
...
         if (!selection.isCollapsed) {
             this.overlay.close();
+        } else if (!selection.inEditable) {
+            const selection = this.document.getSelection();
+            const preventClosing = selection?.anchorNode?.closest?.(
+                "[data-prevent-closing-overlay]"
+            );
+            if (preventClosing?.dataset?.preventClosingOverlay === "true") {
+                return;
+            }
```
Comment (cammarosano) [inline addons/html_editor/static/src/main/link/link_plugin.js]: This does not seems to handle the case where `anchorNode` is a text node.

Comment (cammarosano) [reply addons/html_editor/static/src/main/link/link_plugin.js]: You could do
```js
const preventClosing = !!findNode(closestPath(selection?.anchorNode), (node) => node.dataset?.preventClosingOverlay);
    if (preventClosing) {
           return;
    }
```

Comment (Goaman) [reply addons/html_editor/static/src/main/link/link_plugin.js]: The selection and anchorNode is always defined there, no need to check.
```js
const preventClosing = Boolean(findNode(closestPath(selection.anchorNode), (node) => node.dataset?.preventClosingOverlay));
if (preventClosing) {
return;
}

```

Comment (cammarosano) [reply addons/html_editor/static/src/main/link/link_plugin.js]: I guess he wanted to protect the weird scenarion in which `this.document.getSelection();` returns `null`.
That being said, I don't particularly like creating a `selection` variable in a function where you already have `selection` as a parameter, I find it a bit confusing.

Comment (Jinjiu96) [reply addons/html_editor/static/src/main/link/link_plugin.js]: Thanks for the comments! But I didn't get what you meant about the anchornode is a textnode :confounded: 

Here I'm trying to check when the selection is not in the editable but inside of the linkpopover's editing area (which is the `div` element with `data-prevent-closing-overlay` set to true). So I mean that we just need to check if anchorNode is in the editing `div`, cause if the selection is outside of editable + not in the editing overlay, we close the popover anyways.

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ export class LinkPopover extends Component {
         linkEl: { validate: (el) => el.nodeType === Node.ELEMENT_NODE },
         onApply: Function,
         onRemove: Function,
-        onCopy: Function,
+        closeOverlay: Function,
```
Comment (Goaman) [inline addons/html_editor/static/src/main/link/link_popover.js]: The linktool is not supposed to know that an overlay will close when copying.
So I'd keep the prop `onCopy` and add a prop `onClose`

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 3]
```diff
@@ -95,11 +98,16 @@ export class LinkPopover extends Component {
...
         });
-        this.props.onCopy();
+        this.props.closeOverlay();
     }
     onClickRemove() {
         this.props.onRemove();
     }
+    onClickAway(ev) {
+        if (this.div_editing?.el && !this.div_editing?.el.contains(ev.target)) {
+            this.props.closeOverlay();
```
Comment (Goaman) [inline addons/html_editor/static/src/main/link/link_popover.js]: `this.props.onClose()`

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 4]
```diff
@@ -58,11 +58,14 @@ export class LinkPopover extends Component {
 
         this.keepLastPromise = new KeepLast();
 
+        this.div_editing = useRef("div_editing");
```
Comment (Goaman) [inline addons/html_editor/static/src/main/link/link_popover.js]: this.editingWrapper = useRef("editing-wrapper");

File: addons/html_editor/static/src/main/link/link_popover.xml [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
     <t t-name="html_editor.linkPopover">
         <div class="o-we-linkpopover d-flex bg-white overflow-auto shadow">
-            <div t-if="state.editing" class="m-1 container-fluid d-flex vertical-center">
-                <div class="col m-1">
+            <div t-if="state.editing" class="m-1 container-fluid d-flex vertical-center" t-ref="div_editing">
```
Comment (Goaman) [inline addons/html_editor/static/src/main/link/link_popover.xml]: `editing-wrapper`

File: addons/html_editor/static/src/main/link/link_plugin.js (hunk 1 above)
Comment (Goaman) [reply addons/html_editor/static/src/main/link/link_plugin.js]: @cammarosano the selection here is provided by `onSelectionChange` of the selection pluging and the selection will not be undefined nor null

Comment (cammarosano) [reply addons/html_editor/static/src/main/link/link_plugin.js]: :eyes:  Don't you see a line where `const selection = this.document.getSelection();` is written? :eyes: 
It shadows the `selection` parameter...

Comment (Goaman) [reply addons/html_editor/static/src/main/link/link_plugin.js]: Indeed 🙈

Comment (Jinjiu96) [reply addons/html_editor/static/src/main/link/link_plugin.js]: Yes I had to use `this.document.getSelection();` cause the selection I'd like to check is not in the editable but in the overlay

**Pull request merged**